draft-ietf-gnap-core-protocol.md

```diff
@@ -4777,42 +4777,31 @@ Additional fields can be defined the [Authorization Server Discovery Fields Regi
 
 ## RS-first Method of AS Discovery {#rs-request-without-token}
 
-If the client instance calls an RS without an access token, or with an
-invalid access token, the RS SHOULD respond to the client instance with a
-WWW-Authenticate header field indicating that GNAP needs to be used
-to access the resource. The address of the GNAP
-endpoint MUST be sent in the "as_uri" parameter. The RS MAY
-additionally return a resource reference that the client instance SHOULD use in
-its access token request. This
-resource reference MUST be sufficient for at least the action
-the client instance was attempting to take at the RS and MAY be more
-powerful.
-The means for the RS to determine the resource reference are out of scope
-of this specification, but some dynamic methods are discussed in
+If the client instance calls an RS without an access token, or with an invalid access token, the RS SHOULD be explicit about the fact that GNAP needs to be used to access the resource, by responding with the WWW-Authenticate header field and a GNAP challenge.
+
+In some situations, the client instance might want to know with which specific AS it needs to negotiate for access to that RS.
+The RS MAY additionally return the address of the GNAP endpoint in the "as_uri" parameter, as well as a referer parameter to indicate which RS initiated the discovery process and an opaque reference that the client instance SHOULD both then use in its access token request. The "referer" parameter MUST be the URI of the RS, and the client MUST check its value to protect itself. The opaque reference MUST be sufficient for at least the action the client instance was attempting to take at the RS and MAY be more powerful.
```

Comment:
```suggestion
The RS MAY additionally return the address of the GNAP endpoint in the `as_uri` parameter, a `referrer` parameter to indicate which RS initiated the discovery process, and an opaque `access` reference. The client instance SHOULD then use both the `referrer` and `access` parameters in its access token request. The `referrer` parameter MUST be the URI of the RS, and the client instance MUST check its value to protect itself. The opaque `access` reference MUST be sufficient for at least the action the client instance was attempting to take at the RS and MAY be more powerful.
```

Reply:
Ok, we can revert that. My reasoning here was that access might be a bit misleading, as it actually doesn't open or request an access per say, but only forwards the request